### ISSUE TITLE
ERR_MODULE_NOT_FOUND - api-middleware

### ISSUE THREAD

**Issue by toucdi** [analogjs/analog#831]
### Please provide the environment you discovered this bug in.

node: 18.3
```

"dependencies": {
    "@analogjs/content": "^1.0.0-beta.0",
    "@analogjs/router": "^1.0.0-beta.0",
    "@angular/animations": "^17.0.0",
    "@angular/common": "^17.0.0",
    "@angular/compiler": "^17.0.0",
    "@angular/core": "^17.0.0",
    "@angular/forms": "^17.0.0",
    "@angular/platform-browser": "^17.0.0",
    "@angular/platform-browser-dynamic": "^17.0.0",
    "@angular/platform-server": "^17.0.0",
    "@angular/router": "^17.0.0",
    "@ng-bootstrap/ng-bootstrap": "^16.0.0-rc.2",
    "@nx/angular": "^17.1.0",
    "bootstrap": "^5.3.2",
    "firebase": "^10.7.1",
    "front-matter": "^4.0.2",
    "marked": "^5.0.2",
    "marked-gfm-heading-id": "^3.1.0",
    "marked-highlight": "^2.0.1",
    "mermaid": "^10.2.4",
    "prismjs": "^1.29.0",
    "rxjs": "~7.5.6",
    "tslib": "^2.4.0",
    "zone.js": "~0.14.0"
  },
  "devDependencies": {
    "@analogjs/platform": "^1.0.0-beta.0",
    "@angular-devkit/build-angular": "^17.0.0",
    "@angular/cli": "^17.0.0",
    "@angular/compiler-cli": "^17.0.0",
    "@nx/vite": "^17.1.0",
    "jsdom": "^22.1.0",
    "nx": "^17.1.0",
    "typescript": "~5.2.0",
    "vite": "^4.4.8",
    "vitest": "^0.34.0"
  }
```

### Which area/package is the issue in?

vite-plugin-nitro

### Description

after i have created a analog application with version 1.0.0.beta.2 on windows when i start the application i receive this error:

```
[worker reload] Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'C:\C:\Users\xxxxxx\Documents\projects\beta-test\node_modules\@analogjs\vite-plugin-nitro\src\lib\runtime\api-middleware' imported from C:\Users\xxxxxxxxx\Documents\projects\beta-test\dist\.nitro\dev\index.mjs     
    at new NodeError (node:internal/errors:400:5)
    at finalizeResolution (node:internal/modules/esm/resolve:326:11)
    at moduleResolve (node:internal/modules/esm/resolve:945:10)
    at defaultResolve (node:internal/modules/esm/resolve:1153:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:842:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:77:40)
    at link (node:internal/modules/esm/module_job:76:36)
```

i have checked the file in dist\.nitro\dev\index.mjs and i think there is an errore on import api-middleware:

```
globalThis.importMeta={url:import.meta.url,env:process.env};import { Server } from 'node:http';
// other imports
import { createStorage, prefixStorage } from 'file://C:/Users/l.saija/Documents/projects/whatsapp-channel/node_modules/unstorage/dist/index.mjs';
import unstorage_47drivers_47fs from 'file://C:/Users/l.saija/Documents/projects/whatsapp-channel/node_modules/unstorage/drivers/fs.mjs';
import { toRouteMatcher, createRouter } from 'file://C:/Users/l.saija/Documents/projects/whatsapp-channel/node_modules/radix3/dist/index.mjs';

//wrong import
import _RdX0v2 from '../../../../../../../../../../C:/Users/l.saija/Documents/projects/whatsapp-channel/node_modules/@analogjs/vite-plugin-nitro/src/lib/runtime/api-middleware';
import { initializeApp } from 'file://C:/Users/l.saija/Documents/projects/whatsapp-channel/node_modules/firebase/app/dist/index.mjs';
import { getFirestore, getDocs, collection, doc, getDoc } from 'file://C:/Users/l.saija/Documents/projects/whatsapp-channel/node_modules/firebase/firestore/lite/dist/index.mjs';
```

this reference is used there

```
const handlers = [
  { route: '/v1/channel', handler: _lazy_hsKnGj, lazy: true, middleware: false, method: undefined },
  { route: '', handler: _RdX0v2, lazy: false, middleware: true, method: undefined },
  { route: '/_analog/pages/index', handler: _lazy_cQPQfl, lazy: true, middleware: false, method: undefined },
  { route: '/_analog/pages/channel/:channelId', handler: _lazy_G42F3M, lazy: true, middleware: false, method: undefined }
];
```

can you help me? on macos this issue is not present

### Please provide the exception or error you saw

```true
[worker reload] Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'C:\C:\Users\xxxxxx\Documents\projects\beta-test\node_modules\@analogjs\vite-plugin-nitro\src\lib\runtime\api-middleware' imported from C:\Users\xxxxxxxxx\Documents\projects\beta-test\dist\.nitro\dev\index.mjs     
    at new NodeError (node:internal/errors:400:5)
    at finalizeResolution (node:internal/modules/esm/resolve:326:11)
    at moduleResolve (node:internal/modules/esm/resolve:945:10)
    at defaultResolve (node:internal/modules/esm/resolve:1153:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:842:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:77:40)
    at link (node:internal/modules/esm/module_job:76:36)
```

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [ ] No

**Comment by Saious119** [comment]
I'm also having this issue, its not present on macos as stated and is just a windows issue.

**Comment by sigma390** [comment]
@toucdi  is this issue still open
and if it is open ,ca u assign it to me?

**Comment by monacodelisa** [comment]
yep... a windows issue, 

I am on Windows 11

my first analog app runs fine locally, there is a build error but compiles and runs fine locally 
`package.json`

```
{
  "name": "analog-spartan-app",
  "version": "0.0.0",
  "private": true,
  "engines": {
    "node": ">=18.13.0"
  },
  "scripts": {
    "dev": "ng serve",
    "ng": "ng",
    "start": "npm run dev",
    "build": "ng build",
    "watch": "ng build --watch",
    "test": "ng test"
  },
  "dependencies": {
    "@analogjs/content": "^0.2.0",
    "@analogjs/router": "^0.2.0",
    "@angular/animations": "^17.2.0",
    "@angular/common": "^17.2.0",
    "@angular/compiler": "^17.2.0",
    "@angular/core": "^17.2.0",
    "@angular/forms": "^17.2.0",
    "@angular/platform-browser": "^17.2.0",
    "@angular/platform-browser-dynamic": "^17.2.0",
    "@angular/platform-server": "^17.2.0",
    "@angular/router": "^17.2.0",
    "@nx/angular": "~18.0.0",
    "front-matter": "^4.0.2",
    "marked": "^5.0.2",
    "marked-gfm-heading-id": "^3.1.0",
    "marked-highlight": "^2.0.1",
    "mermaid": "^10.2.4",
    "prismjs": "^1.29.0",
    "rxjs": "~7.5.6",
    "tslib": "^2.4.0",
    "zone.js": "~0.14.0"
  },
  "devDependencies": {
    "@analogjs/platform": "^0.2.0",
    "@angular-devkit/build-angular": "^17.2.0",
    "@angular/cli": "^17.2.0",
    "@angular/compiler-cli": "^17.2.0",
    "@nx/vite": "~18.0.0",
    "nx": "~18.0.0",
    "jsdom": "^22.1.0",
    "typescript": "~5.3.0",
    "vite": "~5.0.0",
    "vitest": "^1.3.1"
  }
}
```
I was making changes to that app which I didn't commit, instead I created a new app for testing 
`package.json`

```
{
  "name": "analog-pure-vscode",
  "version": "0.0.0",
  "type": "module",
  "private": true,
  "engines": {
    "node": ">=18.13.0"
  },
  "scripts": {
    "dev": "ng serve",
    "ng": "ng",
    "start": "npm run dev",
    "build": "ng build",
    "watch": "ng build --watch",
    "test": "ng test"
  },
  "dependencies": {
    "@analogjs/content": "^1.0.0",
    "@analogjs/router": "^1.0.0",
    "@angular/animations": "^17.2.0",
    "@angular/common": "^17.2.0",
    "@angular/compiler": "^17.2.0",
    "@angular/core": "^17.2.0",
    "@angular/forms": "^17.2.0",
    "@angular/platform-browser": "^17.2.0",
    "@angular/platform-browser-dynamic": "^17.2.0",
    "@angular/platform-server": "^17.2.0",
    "@angular/router": "^17.2.0",
    "@nx/angular": "~18.0.0",
    "front-matter": "^4.0.2",
    "marked": "^5.0.2",
    "marked-gfm-heading-id": "^3.1.0",
    "marked-highlight": "^2.0.1",
    "mermaid": "^10.2.4",
    "prismjs": "^1.29.0",
    "rxjs": "~7.5.6",
    "tslib": "^2.4.0",
    "zone.js": "~0.14.0"
  },
  "devDependencies": {
    "@analogjs/platform": "^1.0.0",
    "@analogjs/vite-plugin-angular": "^1.0.0",
    "@angular/cli": "^17.2.0",
    "@angular/compiler-cli": "^17.2.0",
    "@nx/vite": "~18.0.0",
    "nx": "~18.0.0",
    "jsdom": "^22.1.0",
    "typescript": "~5.3.0",
    "vite": "^5.0.0",
    "vitest": "^1.3.1"
  }
}
```
I get the same error 

```
[worker reload] Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'D:\D:\Estee\Programming\My-Projects\analog-testing\analog-pure-vscode-npm\node_modules\@analogjs\vite-plugin-nitro\src\lib\runtime\api-middleware' imported from D:\Estee\Programming\My-Projects\analog-testing\analog-pure-vscode-npm\dist\.nitro\dev\index.mjs
    at new NodeError (node:internal/errors:399:5)
    at finalizeResolution (node:internal/modules/esm/resolve:326:11)
    at moduleResolve (node:internal/modules/esm/resolve:945:10)
    at defaultResolve (node:internal/modules/esm/resolve:1153:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:838:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:77:40)
    at link (node:internal/modules/esm/module_job:76:36)
```
this one is without Tailwind

**Comment by monacodelisa** [comment]
TL;DR - the `dist` folder is missing upon app creation

**Comment by monacodelisa** [comment]
prior the `dist` folder was getting generated on starting the dev server 
`npm start`, `ng serve`, `npm run start` - they were all generating a `dist` folder containing `.nitro` 
now the 'dist` folder is not generated - on windows

**Comment by brandonroberts** [comment]
Thanks everyone. Analog 1.0.2 has been released with the fixes for Windows.
